src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -0,0 +1,185 @@
+package frc.robot.subsystems;
+
+import frc.robot.controllers.PositionDriveController;
+import frc.robot.controllers.SparkMaxSettings;
+import frc.robot.controllers.SparkMaxSettingsBuilder;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import frc.robot.DriveInstruction;
+import frc.robot.RobotConstants;
+import frc.robot.controllers.SparkPositionController;
+import frc.robot.controllers.SparkPositionControllerGroup;
+import frc.robot.path.PositionBuffer;
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANPIDController.AccelStrategy;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.commands.EntechCommandBase;
+import frc.robot.commands.SingleShotCommand;
+import frc.robot.pose.*;
+import frc.robot.utils.EncoderInchesConverter;
+
+public class DriveSubsystem extends BaseSubsystem {
+
+    private boolean inAuto;
+
+    private CANSparkMax frontLeftSpark;
+    private CANSparkMax frontRightSpark;
+    private CANSparkMax rearLeftSpark;
+    private CANSparkMax rearRightSpark;
+
+    private SparkPositionController frontLeftPositionController;
+    private SparkPositionController rearLeftPositionController;
+    private SparkPositionController frontRightPositionController;
+    private SparkPositionController rearRightPositionController;
+
+    private CANEncoder frontLeftEncoder;
+    private CANEncoder frontRightEncoder;
+    private CANEncoder rearLeftEncoder;
+    private CANEncoder rearRightEncoder;
+
+    private SpeedControllerGroup leftSpeedController;
+    private SpeedControllerGroup rightSpeedController;
+    private DifferentialDrive robotDrive;
+    
+    private PositionDriveController autoController;
+    private SparkPositionControllerGroup posController;
+
+    private PositionBuffer positionBuffer = new PositionBuffer();
+
+    private SparkMaxSettings speedSettings = SparkMaxSettingsBuilder.defaults()
+            .withCurrentLimits(35)
+            .coastInNeutral()
+            .withDirections(false, false)
+            .noMotorOutputLimits()
+            .noMotorStartupRamping()
+            .useSpeedControl()
+            .build();
+
+    private SparkMaxSettings positionSettings = SparkMaxSettingsBuilder.defaults()
```

Comment:
do we use both positionSettings and smartMotionSettings? i think we just use the smartMotionOnes...